### DISCUSSION TITLE
E-LOK integration

### DISCUSSION THREAD

**Discussion by dmshimself** [jbergler/hass-ttlock#125] [Q&A]
I see the E-LOK series 9 locks are supported, but I'm having some trouble getting the integration to work.  I've used the E-LOK app to register the lock and that works fine.  I have an approved APP on the TTLock site and the client IDs and secret can be entered fine, but when the workflow asks for the username and password to set up the lock, all I have are those details are for the E-LOK app and they fail.  Do I need to use a TTLock app against this E-LOK and the credentials for that?  I'm concerned doing that might clash with the existing E-LOK app.

**Comment by square-spade** [comment]
Have you tried entering you E-LOK credentials in the integration? I had a quick look and it simply looks like a copy of the TTLock app. If it doesn't work I would just delete the locks from the E-LOK app and use the TTLock app instead, you won't lose any functionality I don't think.

**Reply by dmshimself** [reply]
Some progress.  I found today that during the workflow you need to use the account ID from the TT-LOK app, but the password from the E-LOK app.  The workflow then continued, showed the couple of locks and created a notification for the webhook.  However when I put this into the Open Platform Callback URl, I get an error, which briefly appears about a Java socket error.  I suspect this is because my webhook URL is an http://localhostaddressonmylan, so not an external web address that an external app could access.  

I've refresh the Open Platform page and the local URL I saved isn't stored.  Does my HA need to be externally accessible for this integration to work?

Whoops forgot to mention, the locks appear in HA OK and I can lock/unlock and see the battery levels all OK, so does the webhook need to be in place?

**Reply by jbergler** [reply]
>  Does my HA need to be externally accessible for this integration to work?

For some features, yes.
Without external access for the webhook you can lock/unlock but you won't receive updates when the lock is unlocked or locked physically.

**Reply by dmshimself** [reply]
Many thanks for confirming.  You have far more experience of this integration than me, so do you have any thoughts on how secure providing external access is?  I'm a bit wary.  Is there way way of stopping the notification coming out each time HA is restarted?

**Reply by jbergler** [reply]
The integration makes use of Home Assistants webhook feature. These are protected by randomly generated 'secrets' (in the url). The secret is long enough that it's very very unlikely someone can brute force it. 

That said, you're putting a service on the internet - there's always a risk, but generally as long as you're updating home assistant regularly it's likely fine.

If you pay for Home Assistant cloud, it provides a level of abstraction/security on top of this because they front the webhooks for you but it's not required.

**Comment by dmshimself** [comment]
Very useful. Is there a way to turn off the notification that pops up each
time HA is restarted?
